backend/review/import_utils/import_to_db.py

```diff
@@ -0,0 +1,356 @@
+    ("RREADINGSVALUE", "Readings Value", "readings_value"),
+    ("RAMOUNTLEARNED", "Amount Learned", "amount_learned"),
+    ("RRECOMMENDMAJOR", "Recommend Major", "recommend_major"),
+    ("RRECOMMENDNONMAJOR", "Recommend Non-Major", "recommend_nonmajor"),
+    ("RABILITIESCHALLENGED", "Abilities Challenged", "abilities_challenged"),
+    ("RCLASSPACE", "Class Pace", "class_pace"),
+    ("RINSTRUCTOREFFECTIVE", "Instructor Effectiveness", "instructor_effective"),
+    ("RNATIVEABILITY", "Native Ability", "native_ability"),
+)
+
+# Maps column name from SUMMARY sql tables to common slug.
+COLUMN_TO_SLUG = dict([(x[0], x[2]) for x in REVIEW_BIT_LABEL])
+# Maps "context" value from RATING table to common slug.
+CONTEXT_TO_SLUG = dict([(x[1], x[2]) for x in REVIEW_BIT_LABEL])
+
+
+def titleize(name):
+    """
+    Titleize a course name or instructor, taking into account exceptions such as II.
+    """
+    # string.title() will capitalize the first letter of every word,
+    # where a word is a substring delimited by a non-letter character. So, "o'leary" is two words
+    # and will be capitalized (properly) as "O'Leary".
+    name = name.strip().title()
+    # Roman-numeral suffixes
+    name = re.sub(r"([XVI])(x|v|i+)", lambda m: m.group(1) + m.group(2).upper(), name)
+    # "1st".title() -> "1St", but it should still be "1st".
+    name = re.sub(r"(\d)(St|Nd|Rd|Th)", lambda m: m.group(1) + m.group(2).lower(), name)
+    # Like McDonald.
+    name = re.sub(r"Mc([a-z])", lambda m: "Mc" + m.group(1).upper(), name)
+    # Possessives shouldn't get capitalized.
+    name = name.replace("'S", "'s")
+    return name
+
+
+User = get_user_model()
+
+
+def filter_by_term(rows, semesters, semester_key="TERM"):
+    return [row for row in rows if row[semester_key] in semesters]
+
+
+def import_instructor(pennid, fullname, stat):
+    """
+    Given instructor info do one of a few things:
+    1. Create a new instructor and user object and link the two
+    2. Link together any existing instructors/users that should be linked together assuming the
+       given pennid and fullname belong to the same person.
+    """
+    if pennid is not None:
+        stat("instructors_with_pennid")
+
+        user, user_created = User.objects.get_or_create(
+            pk=int(pennid), defaults={"username": uuid.uuid4()}
+        )
+        if user_created:
+            stat("users_created")
+            user.set_unusable_password()
+            user.save()
+
+        if Instructor.objects.filter(user=user).exists():
+            inst = Instructor.objects.get(user=user)
+            inst_created = False
+        elif Instructor.objects.filter(name=fullname).exists():
+            inst = Instructor.objects.get(name=fullname)
+            inst.user = user
+            inst.save()
+            inst_created = False
+        else:
+            inst = Instructor.objects.create(user=user, name=fullname)
+            inst_created = True
```

Comment:
Can this be simplified with a get_or_create?

Reply:
I tried originally, but the way we want to override properties isn't the same for if there's a matching user or a matching username. If a user matches, we don't want to update the name since the name probably came from Shibboleth. If the name matches then we do want to update the user reference.